src/extension/background-script/actions/ln/request.ts

```diff
@@ -43,14 +44,22 @@ const request = async (
       throw new Error("Could not find an allowance for this host");
     }
 
+    if (!accountId) {
+      throw new Error("Could not find a selected account");
+    }
+
     const connectorName = connector.constructor.name.toLowerCase();
     // prefix method with webln to prevent potential naming conflicts (e.g. with nostr calls that also use the permissions)
     const weblnMethod = `${WEBLN_PREFIX}${connectorName}/${methodInLowerCase}`;
 
     const permission = await db.permissions
       .where("host")
       .equalsIgnoreCase(origin.host)
-      .and((p) => p.method === weblnMethod)
+      .and(
+        (p) =>
+          (!p.accountId || p.accountId === accountId) &&
```

Comment:
I'd say we are just not backwards compatible and simpley also query htere for the account id. 
This makes the code easier 

and please adjust and extend the test for the request method.